InfiniteLoopBookProject/src/com/st/lms/dao/DAOPublisherImpl.java

```diff
@@ -0,0 +1,149 @@
+			String[] parts = publisherLine.split(",");
+			if(Integer.parseInt(parts[ID]) == id) {
+				publisher = new Publisher(CommaDecode(parts[NAME]), Integer.parseInt(parts[ID]), CommaDecode(parts[ADDRESS]), Long.parseLong(parts[PHONE]));
+			}
+		}
+		buffReader.close();
+		return publisher;
+	}
+	
+
+	@Override
+	public void addPublisher(Publisher publisher) throws IOException {
+		String output = "" + publisher.getID() + "," + CommaEncode(publisher.getName()) + "," + CommaEncode(publisher.getAddr()) + "," + publisher.getPhone() + "\n";
+		Files.write(Paths.get(PublisherFile), output.getBytes(), StandardOpenOption.APPEND);
+	}
+	
+	@Override
+	public void replace(int id, Publisher publisher) throws IOException {
+		FileInputStream fin = new FileInputStream(PublisherFile);
+		FileWriter fw = new FileWriter(PublisherTemp, true);
+		BufferedWriter bw = new BufferedWriter(fw);
+		PrintWriter fout = new PrintWriter(bw);
+		BufferedReader buffReader = new BufferedReader(new InputStreamReader(fin));
+		String publisherLine;
+		while((publisherLine = buffReader.readLine()) != null) {
+			String[] parts = publisherLine.split(",");
+			if(Integer.parseInt(parts[ID]) == id) {
+				publisherLine = "" + publisher.getID() + ','
+						+ CommaEncode(publisher.getName()) + ','
+						+ CommaEncode(publisher.getAddr()) + ','
+						+ publisher.getPhone();
+			}
+			fout.println(publisherLine);
+		}
+		buffReader.close();
+		fout.close();
+		File oldFile = new File(PublisherFile);
+		oldFile.delete();
+		File newFile = new File(PublisherTemp);
+		newFile.renameTo(oldFile);
+	}
+	
+	@Override
+	public boolean hasPublisher(int id) throws IOException {
+		FileInputStream fin = new FileInputStream(PublisherFile);
+		BufferedReader buffReader = new BufferedReader(new InputStreamReader(fin));
+		String publisherLine;
+		
+		while((publisherLine = buffReader.readLine()) != null) {
+			String[] parts = publisherLine.split(",");
+			if(Integer.parseInt(parts[ID]) == id) {
+				buffReader.close();
+				return true;
+			}
+		}
+		buffReader.close();
+		return false;
+	}
+
+
+	@Override
+	public void delete(int id) throws IOException{
+		FileInputStream fin = new FileInputStream(PublisherFile);
+		FileWriter fw = new FileWriter(PublisherTemp, true);
+		BufferedWriter bw = new BufferedWriter(fw);
+		PrintWriter fout = new PrintWriter(bw);
+		BufferedReader buffReader = new BufferedReader(new InputStreamReader(fin));
+		String publisherLine;
+		while((publisherLine = buffReader.readLine()) != null) {
+			String[] parts = publisherLine.split(",");
+			if(Integer.parseInt(parts[ID]) != id) {
+				fout.println(publisherLine);
+			}
+		}
+		buffReader.close();
+		fout.close();
+		File oldFile = new File(PublisherFile);
+		oldFile.delete();
+		File newFile = new File(PublisherTemp);
+		newFile.renameTo(oldFile);
+	}
+
+	@Override
+	public void setName(int id, String name) throws IOException {
+		// TODO Auto-generated method stub

```

Comment:
Is there a reason why you have not yet filled in this section?

Reply:
This also happens [right below](https://github.com/STOzaki/InfiniteLoop/blob/b4b4c36303b919901d9f3902947f37dfbf5bab0b/InfiniteLoopBookProject/src/com/st/lms/dao/DAOPublisherImpl.java#L143)

Reply:
Okay, you are right that it is implemented in the super class. But if that is the case, you should get rid of this.